model/game/game.js

```diff
@@ -34,14 +36,15 @@ class Game {
   }
 
   play() {
-    this.deal();
-    let winner;
+    return this.deal().then(() => this.keepPlayingUntilWinner(this.playRound()));
+  }
 
-    // while there are cards and players remain, keep playing
-    while (!(winner = this.declareWinner())) {
-      this.playRound();
-    }
-    this.winner = winner;
+  keepPlayingUntilWinner(roundResult) {
```

Comment:
I used this approach in both the match and the game.  Do you know a better pattern for times when you need to chain promises while waiting on a condition like it were a while loop?